lib/SPIRV/OCLToSPIRV.cpp

```diff
@@ -742,56 +742,28 @@ void OCLToSPIRVBase::visitCallConvert(CallInst *CI, StringRef MangledName,
   if (auto *VecTy = dyn_cast<VectorType>(SrcTy))
     SrcTy = VecTy->getElementType();
   auto IsTargetInt = isa<IntegerType>(TargetTy);
-  auto TargetSigned = DemangledName[8] != 'u';
 
-  std::string TargetTyName(
+  std::string ConversionFunc(
       DemangledName.substr(strlen(kOCLBuiltinName::ConvertPrefix)));
-  auto FirstUnderscoreLoc = TargetTyName.find('_');
-  if (FirstUnderscoreLoc != std::string::npos)
-    TargetTyName = TargetTyName.substr(0, FirstUnderscoreLoc);
 
-  // Validate target type name
-  std::regex Expr("([a-z]+)([0-9]*)$");
+  // Validate target type name and vector size if present
+  std::regex Expr("(float|double|half|u?char|u?short|u?int|u?long)(2|3|4|8|16)*"
+                  "(_sat)*(_rt[ezpn])*$");
   std::smatch DestTyMatch;
-  if (!std::regex_match(TargetTyName, DestTyMatch, Expr))
+  if (!std::regex_match(ConversionFunc, DestTyMatch, Expr))
     return;
 
   // The first sub_match is the whole string; the next
```

Comment:
This comment might need to be rewritten and also appear before the line defining DestTy?

Thanks